### ISSUE TITLE
Weird sassy code create a infinite loop

### ISSUE THREAD

**Issue by MoOx** [cssnext/cssnext#168]
The following code (which is not valid css) seems to create an infinite loop (even on the playground). This code can be parsed by postcss so I guess one plugin is having trouble with it.

Ref https://gitter.im/postcss/postcss?at=559107874a7048d725c22209

```css
/*  Variables - Sass Way */
$blue: blue;

:root {
    /* Variables - CSS Next */
    --red: red;
}

.userinfo {
	.username {
		color: var(--red);
	}

	.numrepos {
		color: $blue;
	}
}


@define-mixin icon $network, $color: blue {
    .icon.is-$(network) {
        color: $color;
        @mixin-content;
    }
    .icon.is-$(network):hover {
        color: white;
        background: $color;
    }
}

@mixin icon twitter {
    font-size: 10px;
}
@mixin icon youtube, red {
    font-size: 12px;
}
```

We should find wtf.

**Comment by MoOx** [comment]
The problem come from postcss-pseudo-class-any-link poke @jonathantneal.
Meanwhile it's fixed in this module, you can disable this features by using this option

```js
{
  features: {
    pseudoClassAnyLink: false,
  }
}
```

From what I can read of the code of the plugin that is responsible, I am starting to think that it can came from a module used by it: postcss-selector-parser poke @ben-eb (because I don't see where the infinite loop might came from)

**Comment by MoOx** [comment]
Related issue: https://github.com/postcss/postcss-selector-parser/issues/12

**Comment by ben-eb** [comment]
Fixed upstream, in release `1.1.2`.

**Comment by MoOx** [comment]
Indeed, fixed. Working in the playground now.
